scripts/apps/authoring/track-changes/suggestions.js

```diff
@@ -0,0 +1,100 @@
+import {
+    editor3DataKeys,
+    getCustomDataFromEditor
+} from 'core/editor3/helpers/editor3CustomData';
+
+function getAllUserIdsFromSuggestions(suggestions) {
+    const users = [];
+
+    suggestions.forEach(({resolutionInfo, suggestionInfo}) => {
+        users.push(resolutionInfo.resolverUserId);
+        users.push(suggestionInfo.author);
+    });
+
+    return users.filter((value, index, self) => self.indexOf(value) === index);
+}
+
+function getUsersAsQueryArray(suggestions) {
+    return getAllUserIdsFromSuggestions(suggestions)
+        .map((key) => ({_id: key}));
+}
+
+function convertUsersArrayToObject(users) {
+    const usersObj = {};
+
+    users.forEach((user) => {
+        usersObj[user._id] = user;
+    });
+
+    return usersObj;
+}
+
+function getUsersFromAPI(api, query) {
+    return api.query('users', {where: JSON.stringify(query)});
+}
+
+function getTypeText(type) {
+    switch (type) {
+    case 'ADD_SUGGESTION':
+        return 'Added';
```

Comment:
here it would need gettext for strings, you can get it injected in controller

Reply:
or you can use the global one, I see you use translate in the template so it's only needed for string extraction

Reply:
I use `{{ getTypeText(type) | translate }}` in the template. Is `gettext` in the controller a better approach?

Reply:
for the translation it doesn't matter much, but we also need to extract strings for tools to translate (like transifex) and when it's scanning the source code it has no idea what `getTypeText(type)` will be, but it can get string from `gettext('Added')`